admin/products_price_manager.php

```diff
@@ -106,54 +103,37 @@
+      $products_date_available = zen_prepare_date($_POST['products_date_available']);
 
-      $specials_date_available = ((!isset($_POST['special_start']) || zen_db_prepare_input($_POST['special_start']) == '') ? '0001-01-01' : zen_date_raw($_POST['special_start']));
-      $specials_expires_date = ((!isset($_POST['special_end']) || zen_db_prepare_input($_POST['special_end']) == '') ? '0001-01-01' : zen_date_raw($_POST['special_end']));
+      $specials_date_available = zen_prepare_date($_POST['specials_date_available']);
+      $specials_expires_date = zen_prepare_date($_POST['specials_expires_date']);
 
-      $featured_date_available = ((!isset($_POST['featured_start']) || zen_db_prepare_input($_POST['featured_start']) == '') ? '0001-01-01' : zen_date_raw($_POST['featured_start']));
-      $featured_expires_date = ((!isset($_POST['featured_end']) || zen_db_prepare_input($_POST['featured_end']) == '') ? '0001-01-01' : zen_date_raw($_POST['featured_end']));
+      $featured_date_available = zen_prepare_date($_POST['featured_date_available']);
+      $featured_expires_date = zen_prepare_date($_POST['featured_expires_date']);
 
       $tmp_value = (isset($_POST['products_price_sorter']) ? zen_db_prepare_input($_POST['products_price_sorter']) : '');
       $products_price_sorter = (!zen_not_null($tmp_value) || $tmp_value == '' || $tmp_value == 0) ? 0 : $tmp_value;
 
       $sql = "UPDATE " . TABLE_PRODUCTS . "
-              SET products_price = :price:,
-                  products_tax_class_id = :taxClass:,
-                  products_date_available = :dateAvailable:,
+              SET products_price = " . (float)$_POST['products_price'] . ",
+                  products_tax_class_id = " . (int)$_POST['products_tax_class_id'] . ",
+                  products_date_available = " . ($products_date_available != 'null' ? "'" . $products_date_available . "'" : $products_date_available) . ",
                   products_last_modified = now(),
-                  products_status = :status:,
-                  products_quantity_order_min = :orderMin:,
-                  products_quantity_order_units = :orderUnits:,
-                  products_quantity_order_max = :orderMax:,
-                  product_is_free = :isFree:,
-                  product_is_call = :isCall:,
-                  products_quantity_mixed = :qtyMixed:,
-                  products_priced_by_attribute = :pricedByAttr:,
-                  products_discount_type = :discType:,
-                  products_discount_type_from = :discTypeFrom:,
-                  products_price_sorter = :discPriceSorter:,
-                  master_categories_id = :masterCatId:,
-                  products_mixed_discount_quantity = :discQty:
+                  products_status = " . (int)$_POST['products_status'] . ",
+                  products_quantity_order_min = " . (int)$_POST['products_quantity_order_min'] . ",
```

Comment:
Huh, my previous attempt at commenting here didn't "stick" it went something like:

wooohhhh, wait a minute, why is products_quantity_order_min, products_quantity_order_units, and at least products_quantity_order_max being forced to an integer??? What about units of less than 1 (0.01 or 0.5)? What about quantities of any of these being partial values? Even say 1.5?

Why are any of these type casts that are covered by $db->bindVars being removed/modified? 

Even the assignment of say products_date_available from using the string casting where if any portion of the string contains `NULL` to the new validation that ignores if say there are spaces around the 'null' and also don't account for capitalization? Further such use of $db->bindVars provides database sanitization for text which is omitted in this new rewrite...

Reply:
You are right, I missed the fact they should be floats. so I will change that. I don't like using the bindvars where it is in my vision unnecessary where a simple cast to integer or float can be done.